decryptor/postgresql/pg_decryptor.go

```diff
@@ -545,7 +545,7 @@ func (proxy *PgProxy) PgDecryptStream(censor acracensor.AcraCensorInterface, dec
 				if decryptor.IsWholeMatch() {
 					// poison record check
 					// check only if has any action on detection
-					if decryptor.GetPoisonCallbackStorage().HasCallbacks() {
+					if decryptor.IsPoisonRecordCheckOn() && decryptor.GetPoisonCallbackStorage().HasCallbacks() {
```

Comment:
I think it should be this way:

- `poison_detect_enable` false: do nothing, no checks, no logs.
- `poison_detect_enable` true: check for poison, if p.r. found – print error log
      - `poison_run_script_file` if set and p.r. found – execute script 
      - `poison_shutdown_enable` if set and p.r. found – shutdown

Reply:
i would remove `decryptor.GetPoisonCallbackStorage().HasCallbacks()` from this `if` statement